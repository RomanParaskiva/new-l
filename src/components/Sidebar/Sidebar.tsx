import React, { useContext } from "react";
import classNames from "classnames";
import { gridContext } from "../../context/gridContext";

export const Sidebar = () => {
    const { start, clear, running, changeSpeed, setGridSize } =
        useContext(gridContext);
    const defaultBtn = "p-2 w-full text-sm rounded-xl uppercase outline-none text-gray-600 hover:scale-105 font-semibold";

    const btnStyle = classNames(
        defaultBtn,
        `${running ? "bg-red-300" : "bg-green-300"}`
    );

    const blueBtn = classNames(defaultBtn, `bg-blue-300`);

    const purpleBtn = classNames(defaultBtn, `bg-purple-300`);
    return (
        <div>
            <div className="mx-auto flex flex-col w-[300px] gap-3 bg-slate-100 rounded-2xl p-3">

                <fieldset className="flex flex-col gap-2">
                    <button onClick={start} className={btnStyle}>
                        {running ? "Стоп" : "Старт"}
                    </button>

                    <button onClick={clear} className={blueBtn}>
                        Сброс
                    </button>
                </fieldset>

                <fieldset className="flex flex-col gap-2">
                    <legend>Скорость</legend>
                    <button
                        onClick={() => changeSpeed(1000)}
                        className={blueBtn}
                    >
                        1
                    </button>

                    <button
                        onClick={() => changeSpeed(500)}
                        className={blueBtn}
                    >
                        1.5
                    </button>

                    <button
                        onClick={() => changeSpeed(200)}
                        className={blueBtn}
                    >
                        2
                    </button>

                </fieldset>

                <fieldset className="flex flex-col gap-2">
                <button
                    disabled={running}
                    className={classNames(
                        purpleBtn,
                        "disabled:bg-gray-400 disabled:text-gray-600"
                    )}
                    onClick={() => setGridSize(50, 50)}
                >
                    50/50
                </button>

                <button
                    disabled={running}
                    className={classNames(
                        purpleBtn,
                        "disabled:bg-gray-400 disabled:text-gray-600"
                    )}
                    onClick={() => setGridSize(30, 30)}
                >
                    30/30
                </button>

                <button
                    disabled={running}
                    className={classNames(
                        purpleBtn,
                        "disabled:bg-gray-400 disabled:text-gray-600"
                    )}
                    onClick={() => setGridSize(100, 100)}
                >
                    100/100
                </button>
                </fieldset>


            </div>
        </div>
    );
};
