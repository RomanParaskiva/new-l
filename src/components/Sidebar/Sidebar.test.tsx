import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

import { Sidebar } from "./Sidebar";

describe("Sidebar tests", () => {
  test("render UI elements", async () => {
    render(<Sidebar />);

    expect(await screen.findByText(/Старт/)).toBeInTheDocument();

    expect(await screen.findByText(/Сброс/)).toBeInTheDocument();

    expect(await screen.findByText("1")).toBeInTheDocument();

    expect(await screen.findByText("1.5")).toBeInTheDocument();

    expect(await screen.findByText("2")).toBeInTheDocument();

    expect(await screen.findByText("50/50")).toBeInTheDocument();
  });

  test("change range Input", async () => {
    render(<Sidebar />);

    const input = screen.getByTestId("rangeInput");
    const label = screen.getByTestId("rangeLabel");
    const clear = await screen.findByText(/Сброс/);

    fireEvent.change(input, { target: { value: 60 } });

    expect(label).toHaveTextContent("60 %");

    fireEvent.click(clear);

    expect(label).toHaveTextContent("0 %");
  });

});
