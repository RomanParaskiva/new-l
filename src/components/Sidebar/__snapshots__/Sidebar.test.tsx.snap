// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar tests renders Sidebar component and get snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="mx-auto flex flex-col w-[300px] gap-3 bg-slate-100 rounded-2xl p-3"
        >
          <fieldset
            class="flex flex-col gap-2"
          >
            <button
              class="p-2 w-full text-sm rounded-xl uppercase outline-none text-gray-600 hover:scale-105 font-semibold bg-green-300"
            >
              Старт
            </button>
            <button
              class="p-2 w-full text-sm rounded-xl uppercase outline-none text-gray-600 hover:scale-105 font-semibold bg-blue-300"
            >
              Сброс
            </button>
          </fieldset>
          <fieldset
            class="flex flex-col gap-2"
          >
            <legend>
              Скорость
            </legend>
            <button
              class="p-2 w-full text-sm rounded-xl uppercase outline-none text-gray-600 hover:scale-105 font-semibold bg-blue-300"
            >
              1
            </button>
            <button
              class="p-2 w-full text-sm rounded-xl uppercase outline-none text-gray-600 hover:scale-105 font-semibold bg-blue-300"
            >
              1.5
            </button>
            <button
              class="p-2 w-full text-sm rounded-xl uppercase outline-none text-gray-600 hover:scale-105 font-semibold bg-blue-300"
            >
              2
            </button>
          </fieldset>
          <fieldset
            class="flex flex-col gap-2"
          >
            <legend>
              Размеры
            </legend>
            <button
              class="p-2 w-full text-sm rounded-xl uppercase outline-none text-gray-600 hover:scale-105 font-semibold bg-purple-300 disabled:bg-gray-400 disabled:text-gray-600"
            >
              50/50
            </button>
            <button
              class="p-2 w-full text-sm rounded-xl uppercase outline-none text-gray-600 hover:scale-105 font-semibold bg-purple-300 disabled:bg-gray-400 disabled:text-gray-600"
            >
              30/30
            </button>
            <button
              class="p-2 w-full text-sm rounded-xl uppercase outline-none text-gray-600 hover:scale-105 font-semibold bg-purple-300 disabled:bg-gray-400 disabled:text-gray-600"
            >
              100/100
            </button>
          </fieldset>
          <fieldset>
            <legend>
              Процент заполнености
            </legend>
            <div
              class="relative pt-1"
            >
              <label
                class="form-label"
                data-testid="rangeLabel"
                for="customRange1"
              >
                0 %
              </label>
              <input
                class="form-range appearance-none w-full h-3 rounded-lg p-0 bg-blue-100 focus:outline-none focus:ring-0 focus:shadow-none"
                data-testid="rangeInput"
                id="customRange1"
                max="80"
                min="0"
                type="range"
                value="0"
              />
            </div>
          </fieldset>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="mx-auto flex flex-col w-[300px] gap-3 bg-slate-100 rounded-2xl p-3"
      >
        <fieldset
          class="flex flex-col gap-2"
        >
          <button
            class="p-2 w-full text-sm rounded-xl uppercase outline-none text-gray-600 hover:scale-105 font-semibold bg-green-300"
          >
            Старт
          </button>
          <button
            class="p-2 w-full text-sm rounded-xl uppercase outline-none text-gray-600 hover:scale-105 font-semibold bg-blue-300"
          >
            Сброс
          </button>
        </fieldset>
        <fieldset
          class="flex flex-col gap-2"
        >
          <legend>
            Скорость
          </legend>
          <button
            class="p-2 w-full text-sm rounded-xl uppercase outline-none text-gray-600 hover:scale-105 font-semibold bg-blue-300"
          >
            1
          </button>
          <button
            class="p-2 w-full text-sm rounded-xl uppercase outline-none text-gray-600 hover:scale-105 font-semibold bg-blue-300"
          >
            1.5
          </button>
          <button
            class="p-2 w-full text-sm rounded-xl uppercase outline-none text-gray-600 hover:scale-105 font-semibold bg-blue-300"
          >
            2
          </button>
        </fieldset>
        <fieldset
          class="flex flex-col gap-2"
        >
          <legend>
            Размеры
          </legend>
          <button
            class="p-2 w-full text-sm rounded-xl uppercase outline-none text-gray-600 hover:scale-105 font-semibold bg-purple-300 disabled:bg-gray-400 disabled:text-gray-600"
          >
            50/50
          </button>
          <button
            class="p-2 w-full text-sm rounded-xl uppercase outline-none text-gray-600 hover:scale-105 font-semibold bg-purple-300 disabled:bg-gray-400 disabled:text-gray-600"
          >
            30/30
          </button>
          <button
            class="p-2 w-full text-sm rounded-xl uppercase outline-none text-gray-600 hover:scale-105 font-semibold bg-purple-300 disabled:bg-gray-400 disabled:text-gray-600"
          >
            100/100
          </button>
        </fieldset>
        <fieldset>
          <legend>
            Процент заполнености
          </legend>
          <div
            class="relative pt-1"
          >
            <label
              class="form-label"
              data-testid="rangeLabel"
              for="customRange1"
            >
              0 %
            </label>
            <input
              class="form-range appearance-none w-full h-3 rounded-lg p-0 bg-blue-100 focus:outline-none focus:ring-0 focus:shadow-none"
              data-testid="rangeInput"
              id="customRange1"
              max="80"
              min="0"
              type="range"
              value="0"
            />
          </div>
        </fieldset>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
